cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -903,16 +903,7 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
         | Opcode::FmaxPseudo => implemented_in_isle(ctx),
 
         Opcode::Icmp => {
-            let condcode = ctx.data(insn).cond_code().unwrap();
-            let dst = get_output_reg(ctx, outputs[0]).only_reg().unwrap();
-            let ty = ctx.input_ty(insn, 0);
-            if ty == types::I128 && condcode != IntCC::Equal && condcode != IntCC::NotEqual {
-                let condcode = emit_cmp(ctx, insn, condcode);
```

Comment:
`emit_cmp` remains used because of the branch cases below, right?

It's worth thinking a bit about how we're going to factor this apart so branches can use the `icmp` logic as well (perhaps it becomes a helper that returns "result in rflags with this CC if possible, otherwise this 0/1 bool in register"), but that's not strictly necessary for this PR. If you're looking for a next thing to do, sketching out the conditional branch lowering (s390x already uses ISLE for this, as a reference point) would be good while this is in your cache :-)